src/main/java/Duke.java

```diff
@@ -1,10 +1,217 @@
+                        break;
+                    case "delete":
+                        textNumber = Integer.parseInt(splitString[1]) - 1;
+                        System.out.println(line);
+                        System.out.println("Noted. I've removed this task:");
+                        System.out.println(tasks.get(textNumber));
+                        System.out.println("Now you have " + (tasks.size() - 1) + " tasks in the list.");
+                        System.out.println("\n" + line);
+                        tasks.remove(textNumber);
+                        break;
+                    case "bye":
+                        break;
+                    default:
+                        throw new InvalidCommandException();
+                }
+                WriteToFile(filePath);
+
+            } catch (ArrayIndexOutOfBoundsException exception) {
+                System.out.println(line);
+                System.out.println("OOPS!!! The description of a " + splitString[0] + " cannot be empty.");
+                System.out.println("\n" + line);
+            } catch (InvalidCommandException exception) {
+                System.out.println(line);
+                System.out.println("OPPS!!! I'm sorry, but I don't know what that means :-(");
+                System.out.println("\n" + line);
+            }
+
+        } while (!commands.equals("bye"));
+
+        System.out.println(line);
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("\n" + line);
+    }
+
+    public static void WriteToFile(String filePath) {
+            try {
+                ClearFile(filePath);
+                CreateFile(filePath);
+                FileWriter myWriter = new FileWriter(filePath);
+                for (int i = 0; i < tasks.size(); i++) {
+                    if(tasks.get(i).isDone == true) {
+                        myWriter.write("1|" +tasks.get(i).toString() + "\n");
+                    }
+                    else{
+                        myWriter.write("0|" +tasks.get(i).toString() + "\n");
+                    }
+                }
+                myWriter.close();
+            } catch (IOException e) {
+                System.out.println("An error occur");
+            }
+    }
+
+    public static void ReadFile(String filePath){
+        try{
+            File myFile = new File(filePath);
+            Scanner myReader = new Scanner(myFile);
+            while (myReader.hasNextLine()) {
+                String file = myReader.nextLine();
+                System.out.println(file);
+                String splitString[] = file.split(" ", 2);
+                if(file.contains("[T]")){
+                    Todo todo = new Todo(splitString[1]);
+                    if(file.contains("1")) {
+                        todo.markAsDone();
+                    }
+                    tasks.add(todo);
+                }
+                else if(file.contains("[D]")) {
```

Comment:
I see that you read the saved file line by line and then check if it contains the correct symbols and numbers. However, will this be a problem if my todo task name has a "1" but it is not marked as done yet.